src/PotentialSimulation/SimulationAlgorithms/SOR.jl

```diff
@@ -24,3 +24,28 @@ end
     end 
     nothing
 end
+
+
+function _guess_optimal_number_of_threads_for_SOR(gs::NTuple{3, Integer}, max_nthreads::Integer, S::Union{Type{Cylindrical}, Type{Cartesian}})::Int
+    max_nthreads = max_nthreads > Base.Threads.nthreads() ? Base.Threads.nthreads() : max_nthreads
+    n = S == Cylindrical ? gs[2] * gs[3] : gs[1] * gs[2] # Number of grid points to be updated in each iteration of the outer loop
+    #=
+        Due to threading overhead calculating small grids with too many threads results in a worse performance.
+        Thus, we limit them depending on the grid size.
+        These ranges may depend on the hardware and are set by educated guesses.
+    =#
+    nt = if n < 100
+        4
+    elseif n < 200
+        8
+    elseif n < 400
+        16
+    elseif n < 800
+        32
+    elseif n < 1600
+        64
+    else
+        max_nthreads
+    end
+    nt < max_nthreads ? nt : max_nthreads
```

Comment:
Compactify lines 37-50 to:
```nt = min(nextpow(2, max(cld(n, 25),4)), max_nthreads)```

Explanation:
- `cld(n,25) = ceil(n / 25)` defines the cases that you use in the `if`-statements here
- `max( ...., 4)` ensures that `nt` is minimum `4`
- `nextpow(2, nt)` find the nearest power of `2` that is greater than `nt`
- `min( ... ,max_nthreads)` ensures that the return value is smaller than `max_nthreads`

![nthreads](https://user-images.githubusercontent.com/30291312/130429067-a114c3e6-bf2e-403a-a8fc-803dbca695a8.png)